pettingzoo/tests/seed_test.py

```diff
@@ -1,6 +1,44 @@
 import warnings
 import random
 import numpy as np
+import hashlib
+import pickle
+
+
+MAX_ENV_ITERS = 50
+
+
+def hash(val):
+    val = pickle.dumps(val)
+    hasher = hashlib.md5()
+    hasher.update(val)
+    return hasher.hexdigest()
+
+
+def calc_hash(new_env, actions, rand_issue=1):
+    cur_hashes = []
+    for i in range(3):
+        new_env.reset()
+        for j in range(rand_issue + 1):
+            random.randint(0, 1000)
+            np.random.normal(size=100)
+        for j in range(MAX_ENV_ITERS):
+            obs, rew, done, info = new_env.last()
+            if done:
+                action = None
+            elif 'legal_moves' in info:
+                basic_action = actions[new_env.agent_selection][j]
+                moves = info['legal_moves']
+                action = moves[basic_action % len(moves)]
+            else:
+                action = actions[new_env.agent_selection][j]
```

Comment:
I don't understand the indexing here. If you want to test every action, why not just set MAX_ENV_ITERS to the size of the action space? If not, why not use the same sampling method you've used in the other tests?

Reply:
Ok, I cleaned up this whole mess. The solution is to seed the action space, and then sampling will be determinstic.